src/containers/community/CommunityJoin.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { connect } from 'react-redux'
+const { func, bool, string } = React.PropTypes
+import { get, isEmpty } from 'lodash'
+import { JOIN_COMMUNITY_WITH_CODE, resetError, joinCommunityWithCode, navigate, resetCommunityValidation, validateCommunityAttribute } from '../../actions'
+import cx from 'classnames'
+
+@connect(({communityValidation, errors}) => {
+  var error
+  if (errors[JOIN_COMMUNITY_WITH_CODE]) {
+    error = 'We couldn\'t find that code, please check it and try again'
+  }
+  let codeInvalid = !get(communityValidation, 'beta_access_code.exists')
```

Comment:
Actually don't want to explicitly check false. Both false and undefined should set codeInvalid to true. This is what let's me omit the codeEmpty stuff entirely.